core/auth/src/main/java/software/amazon/awssdk/auth/credentials/ProfileCredentialsProvider.java

```diff
@@ -127,29 +116,65 @@ public AwsCredentials resolveCredentials() {
         if (loadException != null) {
             throw loadException;
         }
-        return credentialsProvider.resolveCredentials();
+
+        ProfileFile cachedOrRefreshedProfileFile = refreshProfileFile();
+        if (isNewProfileFile(cachedOrRefreshedProfileFile)) {
+            currentProfileFile = cachedOrRefreshedProfileFile;
+            handleProfileFileReload(cachedOrRefreshedProfileFile);
+        }
+
+        return credentials;
```

Comment:
We should cache the provider created in `updateCredentials`, rather than the credentials. This is because depending on the credential provider that's created, the credentials can still change without the profile file changing; e.g. for assume rule.

Reply:
Sounds good. Change made.